openmc/lattice.py

```diff
@@ -1143,13 +1259,55 @@ def get_universe_index(self, idx):
         else:
             return (idx[2], i_ring, i_within)
 
+
+    def _get_universe_index_ox(self, idx):
+        r"""Return index in the universes array corresponding to a lattice element index
+            numeration opposite clockwise
+
+        Parameters
+        ----------
+        idx : Iterable of int
+            Lattice element indices in the :math:`(\alpha,y,z)` coordinate
+            system
+
+        Returns
+        -------
+        2- or 3-tuple of int
+            Indices used when setting the :attr:`HexLattice.universes` property
+
+        """
+
+        # First we determine which ring the index corresponds to.
+        a = idx[0]
+        y = idx[1]
+        z = -a - y
+        g = max(abs(y), abs(a), abs(z))
+
+        # Next we use a clever method to figure out where along the ring we are.
+        i_ring = self._num_rings - 1 - g
+        if y >= 0:
+            if a >= 0:
+                i_within = y
+            else:
+                i_within = 2*g + z
+        else:
+            if a <= 0:
+                i_within = 3*g - y
+            else:
+                i_within = 5*g - z
+
+        if self.num_axial is None:
+            return (i_ring, i_within)
+        else:
+            return (idx[2], i_ring, i_within)
```

Comment:
This function is almost identical to the y case. I would combine them and just conditionally set (x, a) and (a, y) depending on the orientation type. Perhaps you can call them a1 and a2 (as in axis 1 and axis 2). Something like:
```Python
if self._orientation == "x":
    a1, a2 = idx[0], idx[1]
else:
    a1, a2 = idx[1], idx[0]
```